features/webassembly_extensions.md

```diff
@@ -0,0 +1,190 @@
+WebAssembly extensions provide a way to extend Istio proxy's functionality and replaces Mixer as the primary
+extension mechanism in Istio.
+ 
+---
+
+## Experimental
+
+### Requirements:
+
+[//]: # (All information in this section is mandatory for promotion. Please modify the links in this)
+[//]: # (section.)
+
+- [ ] [User stories](insert_your_link_here) reviewed in a work group meeting.
+
+[//]: # (User stories are a way to communicate user value. User stories follow the style)
+[//]: # (as a [type of user], I want [an action] so that [a benefit/a value]. Istio currently has no user)
+[//]: # (story template. Maybe you can make one?)
+
+[//]: # (User stories must be presented in a work group meeting. They need no approval and are later integrated)
+[//]: # (into the RFCs, which do need approval for alpha. You may find value to negotiate within the work group where the)
+[//]: # (user stories are presented to help clarify the user stories.)
+
+- [ ] [RFC Authored] - [create an RFC using template](https://docs.google.com/document/d/1ewJoCcw5-04crH-M0xw4zFxz1cfwVCPnNyW4K3m4Yyc/template/preview).
+
+[//]: # (An RFC is mandatory to graduate to experimental. The RFC does not have to be reviewed in a work group)
+[//]: # (meeting to graduate to experimental.)
+
+- [ ] [Documentation](insert_your_link_here) for enabling and using the feature.
+
+[//]: # (The documentation instructions may exist on the developer wiki or the team drive. They may include instructions)
+[//]: # (for building running a `istioctl experimental command`, or using the preview profile,)
+[//]: # (or any other relevant information.)
+
+- [ ] [Feedback plan](insert_your_link_here).
+
+[//]: # (This may include user feedback meetings, discuss.istio.io conversations, GitHub issues, or mailing lists.)
+
+- [ ] Disabled by default.
+
+- [ ] No impact on performance when the feature is disabled.
+
+---
+
+## Alpha
+
+### Requirements: 
+
+**Design**
+
+- [X] RFC has been approved describing the intention of the feature as well as the user stories behind the feature. 
+
+ [WebAssembly Runtime Validation](https://docs.google.com/document/d/12WUe09imQf6gGgbOZN_vQOcKkd_nek3v7-aGHdNeeSU/edit#)
+ [Istio WebAssembly Plugin SDK](https://docs.google.com/document/d/1_MUmeaUxuL1Ialr_Dvk7vYchDp4GFAtHrLBJnBzODcs/edit#heading=h.xw1gqgyqs5b)
+ [Wasm Extensions API](https://docs.google.com/document/d/1JKg2V86WUftNs5uYCuQ-9X4-JVyF-jM5dpfDyodm5r0/edit#heading=h.xw1gqgyqs5b)
+ [Istio Extensions Configuration API](https://docs.google.com/document/d/1jFuZM4UCKw6AMDmXMIYqbdlyJk95ut2zMPLSNdrUtW8/edit#heading=h.qex63c29z2to)
+
+**Config**
+
+- [X] Explicit user action is required to enable this feature (e.g. a config field, config resource, or installation action). 
+[Istio Ecosystem Wasm Extensions](https://github.com/istio-ecosystem/wasm-extensions)
+> Link to instructions for enabling
+
+**Docs**
+
+- [ ] Reference docs are published to preliminary.istio.io or the Istio wiki.
```

Comment:
I will try to get the existing Wasm doc updated to reflect the new API this week.